crates/ruff_python_semantic/src/model.rs

```diff
@@ -856,6 +865,27 @@ impl<'a> SemanticModel<'a> {
         &self.scopes[self.scope_id]
     }
 
+    /// Returns the parent of the given scope, if any.
+    pub fn scope_parent(&self, scope: &Scope) -> Option<&Scope<'a>> {
+        scope.parent.map(|scope_id| &self.scopes[scope_id])
+    }
+
+    /// Returns the first parent of the given scope that is not a [`ScopeKind::Type`] scope, if any.
+    pub fn scope_parent_skip_types(&self, scope: &Scope) -> Option<&Scope<'a>> {
```

Comment:
Hmm, maybe `non_type_scope_parent`, so that it describes the data returned rather than the action taken? But not a strong opinion.

Reply:
I'm not attached to my name but I don't like that one either haha maybe `scope_non_type_parent`? eek. I also wanted to try `scope_first_parent_not(kind: ScopeKind)` or something but making it generic felt weird too.

Reply:
Maybe what you have here is just clearest... Or `scope_parent_skip_type_scopes`???

Reply:
Maybe `first_non_type_parent_scope` or `non_type_parent_scope`?

Or we name it `first_value_scope`. The way I think about it is that variables can either bind a *value* or a *type*. 

Unrelated: Is this a python thing that types have their own scope? In JS, they all end up in the same scope, but each binding is either a *value* or a *type*.

Reply:
Yeah, surprisingly they added an entirely new lexical scope to support type parameters specifically in PEP 695: https://peps.python.org/pep-0695/#type-parameter-scopes. It's required due to some of the nuances of when these things are evaluated.

Reply:
(Perhaps we should consider a more specific name for that scope kind though.)

Reply:
I kind of like `TypeParam` scope more than `Type` but I didn't want to deviate in this first pass